src/history/hs.c

```diff
@@ -279,6 +279,12 @@ __wt_hs_cursor(WT_SESSION_IMPL *session, WT_CURSOR **cursorp, uint32_t *session_
                 if (!cache->hs_session_inuse[i]) {
                     *cursorp = cache->hs_session[i]->hs_cursor;
                     cache->hs_session_inuse[i] = true;
+                    if (reader) {
+                        cache->hs_session[i]->iface.begin_transaction(
```

Comment:
Borrow the isolation level from the current running transaction. If it's not snapshot then we can skip setting the read timestamp.